homeassistant/components/maico/maico.py

```diff
@@ -0,0 +1,89 @@
+"""Maico Data."""
+from __future__ import annotations
+
+import logging
+
+import httpx
+import xmltodict
+
+from .connection import Connection
+
+# from .sensor import Sensor
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Maico:
+    """Maico Data Class."""
+
+    def __init__(
+        self,
+        device_name,
+        ip,
+        websession: httpx.AsyncClient,
+        username=None,
+        password=None,
+    ):
+        """Initialize Maico object."""
+
+        self._device_name = device_name
+        self._ip = ip
+        self._connection = Connection(websession, self._ip)
+        self._username = username
+        self._password = password
+        self._last_update_time = None
+        self._is_connected = False
+
+        self._sensors: dict = {}
+
+    async def connect(self):
+        """Validate Maico Connection."""
```

Comment:
Validate... how? The caller is evidently supposed to interpret the status code somehow; this should probably raise if the connection fails instead..?